ext/reflection/tests/ReflectionProperty_isReadable.phpt

```diff
@@ -0,0 +1,84 @@
+--TEST--
+Test ReflectionProperty::isReadable()
+--FILE--
+<?php
+
+class A {}
```

Comment:
I would have added some tests for traits. For example, a test ensuring that protected properties from a shared trait are not accessible (https://3v4l.org/hM5TM). Also a trait must not be accepted as a scope.

Reply:
I've seen. I do not agree with changing `isPublic` and friends. It does not accurately reflect the implementation, and I don't think it's even intuitively how people understand it, given the syntax we have chosen. If the syntax were something like `var $prop { public get; private set; }` then it would make a bit more sense. But both syntax and implementation wise, we do not have a get visibility. The get visibility is dictated by the visibility of the entire property. The set visibility can be further restricted.

A BC break is certainly worse than a new feature, so if we decided something needs to be done about this, merging this PR would be much preferable.